File: backend/models/stock.py
from pydantic import BaseModel


class StockSummary(BaseModel):
    code: str
    name: str
    market: str


class StockPrice(BaseModel):
    code: str
    name: str
    currentPrice: float
    changeAmount: float
    changeRate: float
    volume: int
    openPrice: float
    highPrice: float
    lowPrice: float
    prevClosePrice: float


class StockDailyCandle(BaseModel):
    date: str
    openPrice: float
    highPrice: float
    lowPrice: float
    closePrice: float
    volume: int

File: src/types/stock.ts
export interface StockSummary {
  code: string;
  name: string;
  market: string;
}

export interface StockPrice {
  code: string;
  name: string;
  currentPrice: number;
  changeAmount: number;
  changeRate: number;
  volume: number;
  openPrice: number;
  highPrice: number;
  lowPrice: number;
  prevClosePrice: number;
}

export interface StockDailyCandle {
  date: string;
  openPrice: number;
  highPrice: number;
  lowPrice: number;
  closePrice: number;
  volume: number;
}
